modules/common/nats_follower.py

```
@@ -0,0 +1,436 @@
+        if time.monotonic() - self._start_monotonic >= SILENCE_WARN_SECONDS:
+            self._silence_logged = True
+            log.warning("No NATS frames on %s after %.0fs — server reachable "
+                        "but silent (firmware publisher not started?)",
+                        list(self._subjects), SILENCE_WARN_SECONDS)
+
+    # -- background thread -------------------------------------------------
+
+    def _run(self) -> None:
+        try:
+            import nats  # lazy: keeps module import free of the nats-py dep
+        except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)